### ISSUE TITLE
Property 'setMaxListeners' does not exist on type 'WebsocketClient'.ts(2339)

### ISSUE THREAD

**Issue by capaj** [tiagosiebler/binance#440]
incorrect TS types: WebsocketClient instance extends event emitter at runtime, types do not.
This ts-expect-error should not be necessary
![image](https://github.com/user-attachments/assets/b5d07ba3-2de9-492d-9c4c-539539b8d966)

**Comment by tiagosiebler** [comment]
Can you rephrase the question?

**Comment by capaj** [comment]
I tried to rephrase the title. Was that what you meant?

**Comment by tiagosiebler** [comment]
No, the issue you are reporting is not clear to me, maybe I haven't had enough coffee yet today. I have no issues calling setMaxListeners.

**Comment by tiagosiebler** [comment]
Oddly enough, if I test this in the examples folder, I don't have that issue:
<img width="436" alt="Screenshot 2024-08-09 at 10 05 29" src="https://github.com/user-attachments/assets/733469e3-73e4-4484-844c-e68f85a92be5">

I've also tried it in a separate project, with the latest version of this SDK imported, and no issue there either - runs without issues:
<img width="502" alt="Screenshot 2024-08-09 at 10 08 32" src="https://github.com/user-attachments/assets/97845370-30f1-4a77-bf5a-eb565a816e03">

No build warnings either...

**Comment by tiagosiebler** [comment]
Could you reproduce this issue in a really tiny ts project?

**Comment by capaj** [comment]
I am using bun, this is my tsconfig

```
{
  "compilerOptions": {
    "module": "ESNext",
    "esModuleInterop": true,
    "target": "ESNext",
    "moduleResolution": "Bundler",
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "isolatedModules": false,
    "strict": true,
    "noImplicitAny": true,
    "noEmit": true,
    "useUnknownInCatchVariables": false,
    "skipLibCheck": true,
    "inlineSourceMap": true,
    "resolveJsonModule": true,
    "types": ["bun-types"]
  },

  "exclude": ["dashboard/"],
  "lib": ["esnext", "es2021"]
}
```


that could mess up with those event listener types, will try to create a repro

**Comment by tiagosiebler** [comment]
Interesting, could also be bun itself - would be curious to understand what's happening here and how to resolve it. Ideal would be to create a really tiny project with no real code in it to reproduce it - just the few lines causing this and the minimal boilerplate required (such as the tsconfig) to reproduce it. Then we start chopping away until we find what's causing it.

**Comment by Malove98** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by tiagosiebler** [comment]
Closing this until more info is available to reproduce
